src/components/inputs/localized-text-input/localized-text-input.js

```diff
@@ -116,173 +116,174 @@ const RequiredValueErrorMessage = () => (
 
 RequiredValueErrorMessage.displayName = 'RequiredValueErrorMessage';
 
-export default class LocalizedTextInput extends React.Component {
-  static displayName = 'LocalizedTextInput';
+const LocalizedTextInput = props => {
+  let defaultExpansionState = false;
 
-  static RequiredValueErrorMessage = RequiredValueErrorMessage;
+  if (props.hideLanguageExpansionControls) {
+    defaultExpansionState = true;
+  } else if (props.defaultExpandLanguages) {
+    defaultExpansionState = true;
+  }
 
-  static propTypes = {
-    id: PropTypes.string,
-    name: PropTypes.string,
-    autoComplete: PropTypes.string,
-    // then input doesn't accept a "languages" prop, instead all possible
-    // languages have to exist (with empty or filled strings) on the value:
-    //   { en: 'foo', de: '', es: '' }
-    value: PropTypes.objectOf(PropTypes.string).isRequired,
-    onChange: requiredIf(PropTypes.func, props => !props.isReadOnly),
-    selectedLanguage: PropTypes.string.isRequired,
-    onBlur: PropTypes.func,
-    onFocus: PropTypes.func,
-    hideLanguageExpansionControls: PropTypes.bool,
-    defaultExpandLanguages: (props, propName, componentName, ...rest) => {
-      if (
-        props.hideLanguageExpansionControls &&
-        typeof props[propName] === 'boolean'
-      ) {
-        throw new Error(
-          oneLine`
-            ${componentName}: "${propName}" does not have any effect when
-            "hideLanguageExpansionControls" is set.
-          `
-        );
-      }
-      return PropTypes.bool(props, propName, componentName, ...rest);
-    },
-    isAutofocussed: PropTypes.bool,
-    isDisabled: PropTypes.bool,
-    isReadOnly: PropTypes.bool,
-    placeholder: PropTypes.objectOf(PropTypes.string),
-    horizontalConstraint: PropTypes.oneOf(['xs', 's', 'm', 'l', 'xl', 'scale']),
-    hasError: PropTypes.bool,
-    errors: PropTypes.objectOf(PropTypes.node),
-  };
+  const [areLanguagesExpanded, setAreLanguagesExpanded] = React.useState(
+    defaultExpansionState
+  );
 
-  static defaultProps = {
-    horizontalConstraint: 'scale',
-  };
+  const languages = sortLanguages(
+    props.selectedLanguage,
+    Object.keys(props.value)
+  );
 
-  static getId = getId;
+  const toggleLanguages = React.useCallback(() => {
+    setAreLanguagesExpanded(!areLanguagesExpanded);
+  }, [areLanguagesExpanded]);
 
-  static getName = getName;
+  const id = getFieldId(props, {}, sequentialId);
 
-  static createLocalizedString = createLocalizedString;
+  const hasErrorInRemainingLanguages =
+    props.hasError ||
+    getHasErrorOnRemainingLanguages(props.errors, props.selectedLanguage);
 
-  static isEmpty = isEmpty;
+  if (hasErrorInRemainingLanguages) {
+    const shouldLanguagesExpand =
+      hasErrorInRemainingLanguages ||
+      props.hideLanguageExpansionControls ||
+      areLanguagesExpanded;
 
-  static omitEmptyTranslations = omitEmptyTranslations;
+    if (shouldLanguagesExpand !== areLanguagesExpanded) {
```

Comment:
Let's add the link to the docs here, otherwise it might be confusing to see updating the state on rendering.